src/pages/index.js

```diff
@@ -56,28 +74,123 @@ const enableValidation = (config) => {
   });
 };
 enableValidation(config);
-
 /**============================================
  *               Functions
  *=============================================**/
+function handleApiCardInfo(cardData) {
+  const cardApiSection = new Section(
+    { items: cardData, renderer: renderCard },
+    cardContainer
+  );
+  cardApiSection.renderItems();
+}
 function renderCard(item, method = "append") {
-  const card = new Card(item, cardSelector, handleImageClick);
+  const card = new Card(
+    item,
+    cardSelector,
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  );
   const cardElement = card.renderCard();
   cardSection.addItem(cardElement, method);
 }
 function handleImageClick(data) {
   popupWithImage.open(data);
 }
-function handleProfileSubmit(userData) {
+function handleDeleteClick(cardData) {
+  deleteConfirmation.open(cardData);
+}
+function deleteConfirmed(card) {
+  api
+    .deleteCard(card.cardID)
+    .then(() => {
+      card.cardElement.remove();
+      card.cardElement = null;
+    })
+    .then(() => {
+      deleteConfirmation.close();
+    })
+    .catch(console.error);
+}
+function handleLikeClick(cardObj, cardData, likeAction) {
+  if (likeAction === "addLike") {
+    api
+      .addLike(cardData._id)
+      .then(() => {
+        card.addLike(cardObj);
+      })
+      .catch(console.error);
+  }
+  if (likeAction === "removeLike") {
+    api
+      .deleteLike(cardData._id)
+      .then(() => {
+        card.removeLike(cardObj);
+      })
+      .catch(console.error);
+  }
+}
+
+function getInitialApiUserInfo(userData) {
   userInfo.setUserInfo({
-    name: userData.title,
-    description: userData.description,
+    name: userData.name,
+    description: userData.about,
   });
+  userInfo.setUserAvatar(userData.avatar);
+}
+
+function handleProfileSubmit(userData) {
+  profilePopupForm.renderLoading(true, "Saving...");
+  api
+    .setUserInfo(userData.title, userData.description)
+    .then(() => {
+      userInfo.setUserInfo({
+        name: userData.title,
+        description: userData.description,
+      });
+    })
+    .then(() => {
+      profilePopupForm.close();
+      profilePopupForm.popupFormReset();
+    })
+    .catch(console.error)
+    .finally(() => {
+      profilePopupForm.renderLoading(false);
+    });
+}
+function handleAvatarSubmit(data) {
+  avatarPopupForm.renderLoading(true, "Saving...");
```

Comment:
**Could be improved**

`"Saving..."` is not needed here because it's stated by default in the method


```js
  avatarPopupForm.renderLoading(true);


```

Reply:
Removed additional text which is unnecessary.